cluster.go

```diff
@@ -406,14 +454,19 @@ func (c *Cluster) doInner(a Action, addr, key string, ask bool, attempts int) er
 	thisA := a
 	if ask {
 		thisA = WithConn(key, func(conn Conn) error {
-			return askConn{conn}.Do(a)
+			err = askConn{conn}.Do(a)
+			c.traceGotResponse(addr, key, attempts, err)
+			return err
 		})
 	}
 
 	if err = p.Do(thisA); err == nil {
+		c.traceGotResponse(addr, key, attempts, nil)
```

Comment:
You could instead to 

```go
err = p.Do(thisA)
c.traceGotResponse(addr, key, attempts, err)
if err != nil {
    return nil
}
```

and remove the call to `traceGotResponse` on line 468.

Reply:
we could change this line.
```golang
if err == nil {  // changed condition
    return nil
}
```